uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -0,0 +1,516 @@
+define([
+  'leave-absences/shared/modules/controllers',
+  'common/lodash',
+  'common/moment',
+  'common/services/api/option-group',
+  'common/services/hr-settings',
+], function (components, _, moment) {
+  'use strict';
+
+  components.controller('LeaveRequestPopupCtrl', [
+    '$log', '$q', '$rootScope', '$uibModalInstance', 'AbsencePeriod', 'AbsenceType',
+    'api.optionGroup', 'baseData', 'Calendar', 'Entitlement', 'HR_settings',
+    'LeaveRequest', 'LeaveRequestInstance', 'PublicHoliday',
+    function ($log, $q, $rootScope, $modalInstance, AbsencePeriod, AbsenceType,
+      OptionGroup, baseData, Calendar, Entitlement, HR_settings,
+      LeaveRequest, LeaveRequestInstance, PublicHoliday
+    ) {
+      $log.debug('LeaveRequestPopupCtrl');
+
+      var $ctrl = {},
+        serverDateFormat = 'YYYY-MM-DD';
+
+      $ctrl.absenceTypes = [];
+      $ctrl.calendar = {};
+      $ctrl.leaveRequestDayTypes = [];
+      $ctrl.balance = {
+        opening: 0,
+        change: {
+          amount: 0,
+          breakdown: []
+        },
+        closing: 0
+      };
+      $ctrl.period = {};
+      $ctrl.pagination = {
+        totalItems: $ctrl.balance.change.breakdown.length,
+        filteredbreakdown: $ctrl.balance.change.breakdown,
+        currentPage: 1,
+        numPerPage: 5,
+        pageChanged: function () {
+          //filter breakdowns
+          var begin = ((this.currentPage - 1) * this.numPerPage),
+            end = begin + this.numPerPage;
+
+          this.filteredbreakdown = $ctrl.balance.change.breakdown.slice(begin, end);
+        }
+      };
+      $ctrl.uiOptions = {
+        showDatePickerFrom: false,
+        showDatePickerTo: false,
```

Comment:
too many dots in object and moreover a separate object is required to set `datePickerOptions`